bindings_ffi/src/lib.rs

```diff
@@ -128,6 +152,317 @@ impl FfiXmtpClient {
+            start_time_ns: req.start_time_ns,
+            end_time_ns: req.end_time_ns,
+            paging_info: req.paging_info.map(|paging_info| {
+                return FfiPagingInfo {
+                    limit: paging_info.limit,
+                    direction: FfiSortDirection::from_i32(paging_info.direction),
+                    cursor: proto_cursor_to_ffi(paging_info.cursor),
+                };
+            }),
+        }
+    }
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiV2QueryResponse {
+    envelopes: Vec<FfiEnvelope>,
+    paging_info: Option<FfiPagingInfo>,
+}
+
+impl From<QueryResponse> for FfiV2QueryResponse {
+    fn from(resp: QueryResponse) -> Self {
+        Self {
+            envelopes: resp.envelopes.into_iter().map(|env| env.into()).collect(),
+            paging_info: resp.paging_info.map(|paging_info| {
+                return FfiPagingInfo {
+                    limit: paging_info.limit,
+                    direction: FfiSortDirection::from_i32(paging_info.direction),
+                    cursor: None,
+                };
+            }),
+        }
+    }
+}
+
+impl From<FfiV2QueryResponse> for QueryResponse {
+    fn from(resp: FfiV2QueryResponse) -> Self {
+        Self {
+            envelopes: resp.envelopes.into_iter().map(|env| env.into()).collect(),
+            paging_info: resp.paging_info.map(|paging_info| {
+                return PagingInfo {
+                    limit: paging_info.limit,
+                    direction: paging_info.direction as i32,
+                    cursor: None, // TODO: fix me
+                };
+            }),
+        }
+    }
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiV2BatchQueryRequest {
+    requests: Vec<FfiV2QueryRequest>,
+}
+
+#[derive(uniffi::Record)]
+pub struct FfiV2BatchQueryResponse {
+    responses: Vec<FfiV2QueryResponse>,
+}
+
+impl From<BatchQueryRequest> for FfiV2BatchQueryRequest {
+    fn from(req: BatchQueryRequest) -> Self {
+        Self {
+            requests: req.requests.into_iter().map(|req| req.into()).collect(),
+        }
+    }
+}
+
+impl From<FfiV2BatchQueryRequest> for BatchQueryRequest {
+    fn from(req: FfiV2BatchQueryRequest) -> Self {
+        Self {
+            requests: req.requests.into_iter().map(|req| req.into()).collect(),
+        }
+    }
+}
+
+impl From<BatchQueryResponse> for FfiV2BatchQueryResponse {
+    fn from(resp: BatchQueryResponse) -> Self {
+        Self {
+            responses: resp.responses.into_iter().map(|resp| resp.into()).collect(),
+        }
+    }
+}
+
+impl From<FfiV2BatchQueryResponse> for BatchQueryResponse {
+    fn from(resp: FfiV2BatchQueryResponse) -> Self {
+        Self {
+            responses: resp.responses.into_iter().map(|resp| resp.into()).collect(),
+        }
+    }
+}
+
+#[derive(uniffi::Object)]
+pub struct FfiV2Client {
+    inner_client: Arc<xmtp_api_grpc::grpc_api_helper::Client>,
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl FfiV2Client {
```

Comment:
We might want to call this a `FfiV2ApiClient` in case we ever want something like a full V2 client in Rust. Or at least for clarity